rdmd.d

```diff
@@ -694,10 +714,13 @@ private string[string] getDependencies(string rootModule, string workDir,
     immutable rootDir = dirName(rootModule);
 
     // Collect dependencies
-    auto depsGetter =
-        // "cd "~shellQuote(rootDir)~" && "
-        [ compiler ] ~ compilerFlags ~
-        ["-v", "-o-", rootModule, "-I"~rootDir];
+    auto depsGetter = [ compiler ] ~ compilerFlags ~ [
+        "-v",
+        "-c",
+        "-of"~buildPath(objDir, "rdmd.root"~objExt),
```

Comment:
> I used exeBasename as the filename is shown in the error message

Seems like a good idea. Done.

The problem I had worked around here was that dmd would also write to foo.o when -main is passed in the linking command.

I've now changed it to pass -main when compiling the root module. But there seems to be a bug in dmd with `-main -c -of`: [issue 16440](https://issues.dlang.org/show_bug.cgi?id=16440). So I've reverted to rdmd's old style of doing --main: stubmain.d.